File: easyOneLeetCodeR.py
class Solution(object):
  def romanToInt(self, s):
  """
        :type s: str
        :rtype: int
        """
result = 0
mappings = {
  'I': 1,
  'V': 5,
  'X': 10,
  'L': 50,
  'C': 100,
  'D': 500,
  'M': 1000
}
for i in range(len(s)):
  if i+1 < len(s) and mappings[s[i]] < mappings[s[i+1]]:
  result -= mappings[s[i]]
else:
  result += mappings[s[i]]
return result

File: easyOneLeetCodeR.r

romanToInt <- function(s) {
  mappings <- list(
    I = 1,
    V = 5,
    X = 10,
    L = 50,
    C = 100,
    D = 500,
    M = 1000
  )
  result <- 0
  for (i in 1:nchar(s)) {
    if (i+1 < nchar(s) && mappings[[s[i]]] < mappings[[s[i + 1]]]) {
      result <- result - mappings[[s[i]]]
    } else {
      result <- result + mappings[[s[i]]]
    }
  }
  return(result)
}